### DISCUSSION TITLE
How to take the result of one request as an argument in the second request, and use the original HTTP request mode

### DISCUSSION THREAD

**Discussion by ltfafei** [projectdiscovery/nuclei-templates#6877] [Q&A] [answered]
When compiling the Nuclei POC, the first request turns to GET, the URL is http://test1.com, and the response result is: alm_admin_nonce":"1723d297e6","select_authors ", there is a second request, whose request method is GET, and which needs to be added into the request parameter 1723d297e6 in the request result of the first request. To find 1723d297e6 in the result of a request or as the parameter in the second request, give the specific POC code or note that the workflow function of the Nuclei cannot be used and the request contains the variable Cookie: {{cookie}}

Bing AI answer:
```
id: example-poc
info:
  name: Example POC
  author: pdteam
  severity: info

requests:
  - raw: # first request in raw format 
      - |
        GET / HTTP/1.1
        Host: test1.com
    matchers-condition: and
    matchers:
      - type: word # check if the response contains the word test=123
        words:
          - "test=123"
        part: body
    extractors:
      - type: regex # extract the value of test from the response body 
        part: body 
        regex:
          - "test=(\d+)"
        group: 1 # use the first capture group as the extracted value 
        name: test_value # assign a name to the extracted value 

  - raw: # second request in raw format 
      - |
        GET /?test={{test_value}} HTTP/1.1 # use the extracted value as a parameter 
        Host: {{Hostname}} # use the built-in variable for hostname 
    matchers-condition: and 
    matchers:
      - type: status # check if the status code is 200 
        status:
          - 200 
```

Because of the inclusion of the variable cookies, it seems that only raw HTTP requests can be used.

However, after testing, only the first request is executed and the second request is not executed

**Comment by ehsandeep** [comment, marked as answer]
See for example - https://github.com/projectdiscovery/nuclei-templates/blob/5f10649d0fcf016d550bd87d40635809910ea13f/vulnerabilities/other/yapi-rce.yaml

All requests must be under the same request block to extract and share between other requests.

also please use [GitHub discussion](https://github.com/projectdiscovery/nuclei-templates/discussions/categories/q-a) for asking questions instead of creating an issue.
